service/DevHome.Service/ComHelpers.cs

```diff
@@ -88,22 +88,33 @@ public static void VerifyCallerIsInTheSameDirectory()
         // Our caller is in the same directory that we are
     }
 
-    public static void VerifyCallerIsFromTheSamePackage()
+    public static void VerifyCallerIsFromTheSamePackage(Process caller)
     {
         unsafe
         {
             string devHomeServicePackage = Package.Current.Id.FullName;
-            PInvoke.CoImpersonateClient().ThrowOnFailure();
 
-            WindowsIdentity identity = WindowsIdentity.GetCurrent();
+            SafeProcessHandle h = new(PInvoke.OpenProcess(PROCESS_ACCESS_RIGHTS.PROCESS_QUERY_LIMITED_INFORMATION, false, (uint)caller.Id), true);
+
+            if (h.IsInvalid)
+            {
+                throw new UnauthorizedAccessException();
+            }
+
+            bool f = PInvoke.OpenProcessToken(h, TOKEN_ACCESS_MASK.TOKEN_QUERY, out SafeFileHandle token);
+
+            if (!f || token.IsInvalid)
+            {
+                throw new UnauthorizedAccessException();
```

Comment:
We should add logs to see how many UnauthorizedAccessExceptions we see. Maybe not in this PR, but as a future task.

Reply:
It might not be interesting to log it from the service (what would we do if that number spikes? Is it a product failure, or unauthorized folks trying to connect to the service?).

We should definitely log on the client side when we're not able to connect. That I think would be much more interesting. I plan to do that in a future checkin.